File: 217. Contains Duplicate/main.go
package main

import "fmt"

func containsDuplicate(nums []int) bool {

	dupMap := map[int]bool{}

	for _, num := range nums {

		if _, ok := dupMap[num]; ok {
			return true
		} else {
			dupMap[num] = true
		}

	}

	return false

}

func main() {

	nums := []int{1, 2, 3, 3, 6}

	answer := containsDuplicate(nums)

	fmt.Print(answer)

}

File: 217. Contains Duplicate/main.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        number_set = set()

        for n in nums:
            if n not in number_set:
                number_set.add(n)
            else:
                return True
        return False


solution = Solution()
answer = solution.containsDuplicate([1, 2, 3])
print(answer)
